taskvine/src/worker/vine_worker.c

```diff
@@ -520,14 +541,25 @@ static int start_process(struct vine_process *p, struct link *manager)
 		vine_gpus_allocate(t->resources_requested->gpus, t->task_id);
 	}
 
+	/* Starting a function call is different from starting a standard task, so we return early. */
+	if (t->needs_library) {
+		itable_insert(procs_running, p->task->task_id, p);
+		debug(D_VINE, "started function call %d: %s", p->task->task_id, p->task->command_line);
+		p->library_process->functions_running++;
+		return start_function(p);
```

Comment:
If `start_function` fails, does `functions_running` have the right value?

Reply:
No. Fixed!